go/sys/desktop/sessions.go

```
@@ -0,0 +1,243 @@
+//     headless server, etc.)
+//
+// Returns an error only when loginctl is present but its output is
+// malformed or the per-session detail probe fails for a reason other
+// than the session disappearing mid-call.
+func ActiveSessions(ctx context.Context) ([]Session, error) {
+	if _, err := exec.LookPath(loginctlPath); err != nil {
+		// Host doesn't ship systemd-logind. Treat as "no sessions" so
+		// the caller's empty-set policy (skip-with-warn vs fail) drives
+		// the user-facing behavior consistently regardless of the
+		// underlying init system.
+		return nil, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 78) but it returns nil

(nilerr)